src/parallelComm.C

```diff
@@ -111,32 +111,37 @@ void parallelComm::sendRecvPacketsAll(PACKET *sndPack, PACKET *rcvPack)
     if (rcvPack[i].nints > 0) {
       rcvPack[i].intData=(int *) malloc(sizeof(int)*rcvPack[i].nints);
     }
-    if (rcvPack[i].nreals > 0) {
-      rcvPack[i].realData=(REAL *) malloc(sizeof(REAL)*rcvPack[i].nreals);
-    }
   }
```

Comment:
I don't think it's a "bug". In the sense that `rcvPack[i].nreals` is populated after the alltoall on line 47. So it is totally valid to do the malloc here. However, it is also valid to move it later. It's not clear if it is worth doing it before the wait or after the wait. Before the wait means that procs could start populating the struct while waiting on MPI data. Which might be a case for moving the int one (`rcvPack[i].intData=(int *) malloc(sizeof(int)*rcvPack[i].nints);`) earlier. Or maybe it's better to move it after the wait as you do here. TBH I doubt we will see a difference.